mathsymbols.py

```diff
@@ -4992,6 +4992,20 @@ def symbol_by_name(name, exclude=None):
         return symbol_by_name(synonyms[name], exclude)  # synonym may ref on other synonym
     return None
 
+def symbol_by_unique_prefix(prefix):
+    """
+    If the given string is a prefix of exactly one name or synonym, return the
+    associated symbol, otherwise None
+    """
+
+    # determine whether there are 2+ options without generating everything
+    gen_maths = (k for k in maths if k.startswith(prefix))
+    gen_syn = (k for k in synonyms if k.startswith(prefix))
+    options = (k for gen in [gen_maths, gen_syn] for k in gen)
```

Comment:
This will have linear complexity and might be slow. Maybe sort (somewhere before, only once) `maths` & `symbols` and use [bisect](https://docs.python.org/3/library/bisect.html) to find prefix?

Reply:
With `bisect` you can also look-ahead to see if there're longer names with same prefix

Reply:
Addressed in https://github.com/mvoidex/UnicodeMath/pull/37/commits/9fc019c3553795787b3bbcabe24f2df7d64af3d8.